Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/MarkerVisualDetectorSpatialLocalizer.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.MixedReality.Experimental.SpatialAlignment.Common;
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.MarkerDetection;
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView.Utilities;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
+{
+    internal class MarkerVisualDetectorLocalizationSettings : ISpatialLocalizationSettings
+    {
+        public MarkerVisualDetectorLocalizationSettings() { }
+
+        public void Serialize(BinaryWriter writer) { }
+    }
+
+    internal class MarkerVisualDetectorSpatialLocalizer : SpatialLocalizer<MarkerVisualDetectorLocalizationSettings>
+    {
+        [Tooltip("The reference to an IMarkerDetector GameObject")]
+        [SerializeField]
+        private MonoBehaviour MarkerDetector = null;
+        private IMarkerDetector markerDetector = null;
+
+        public override Guid SpatialLocalizerId => Id;
+        public static readonly Guid Id = new Guid("2DA7D277-323F-4A0D-B3BB-B2BA6D3EF70E");
+
+#if UNITY_EDITOR
+        private void OnValidate()
+        {
+            FieldHelper.ValidateType<IMarkerDetector>(MarkerDetector);
+        }
+#endif
+
+        /// <inheritdoc />
+        public override bool TryCreateLocalizationSession(IPeerConnection peerConnection, MarkerVisualDetectorLocalizationSettings settings, out ISpatialLocalizationSession session)
+        {
+            markerDetector = (markerDetector == null) ? MarkerDetector as IMarkerDetector : markerDetector;
+            session = new LocalizationSession(this, settings, peerConnection, debugLogging);
+            return true;
+        }
+
+        /// <inheritdoc />
+        public override bool TryDeserializeSettings(BinaryReader reader, out MarkerVisualDetectorLocalizationSettings settings)
+        {
+            settings = new MarkerVisualDetectorLocalizationSettings();
+            return true;
+        }
+
+        private class LocalizationSession : DisposableBase, ISpatialLocalizationSession
+        {
+            private readonly MarkerVisualDetectorSpatialLocalizer localizer;
+            private readonly MarkerVisualDetectorLocalizationSettings settings;
+            private readonly IPeerConnection peerConnection;
+            private readonly ISpatialCoordinateService coordinateService;
```

Comment:
>ISpatialCoordinateService [](start = 29, length = 25)

I think if this field was a MarkerDetectorCoordinateService instead of an ISpatialCoordinateService, you'd be able to use the overloads that know the coordinate ID type is 'int' and not 'string'.